Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "FixURDF.h"
+#include <AzCore/XML/rapidxml_print.h>
+#include <AzCore/std/containers/unordered_set.h>
+#include <iostream>
+
+namespace ROS2::Utils
+{
+    AZStd::vector<AZStd::string> AddMissingInertiaToLinks(AZ::rapidxml::xml_node<>* urdf)
+    {
+        AZStd::vector<AZStd::string> modifiedLinks;
+        using namespace AZ::rapidxml;
+        AZStd::vector<xml_node<>*> links;
+        for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
+        {
+            if (!link->first_node("inertial"))
+            {
+                links.push_back(link);
+            }
+        }
+
+        for (auto* link : links)
+        {
+            xml_node<>* inertial = urdf->document()->allocate_node(node_element, "inertial");
+
+            xml_node<>* mass = urdf->document()->allocate_node(node_element, "mass");
+            mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
```

Comment:
According to the URDF spec, this should actually default to a 0 mass if not specified: http://wiki.ros.org/urdf/XML/link . "<inertial> (optional: defaults to a zero mass and zero inertia if not specified)". I know the SDF converter requires this to be specified, but does the SDF converter also require this to be 1 and identity for inertia to avoid the error?

Reply:
I would not set it zero, since it will cause serious problems down the line in PhysX. I think unit mass and identity inertia is reasonable choice.